### DISCUSSION TITLE
Grails 7 Scope - Should Grails 7 update to Spring Boot 3.3?

### DISCUSSION THREAD

**Discussion by sdelamo** [apache/grails-core#13518] [Polls]
This is a poll to pulse the community about the Grails 7 scope. In particular about the Spring Boot version to based on.

**Comment by lei-fastspring** [comment]
3.3 is a must I say

**Comment by jamesfredley** [comment]
I wish Boot 2.7 was still supported by Spring, since Boot 3.0+ requires Hibernate 6.5, which requires a material amount of work on GORM as the legacy Hibernate Criteria API was removed in favor of JPA Criteria API.

But that is not in the cards.  Even Spring Boot 3.0 and 3.1 are already out of support and 3.2 support ends 23 Nov 2024. A new Boot version comes every 6 months and receives 12 months of support. https://endoflife.date/spring-boot

**Comment by HaroldHormaechea** [comment]
Considering not only Spring Boot 2.7 is out of support, but Spring 5 itself is going to be [too soon](https://spring.io/blog/2024/03/01/support-timeline-announcement-for-spring-framework-6-0-x-and-5-3-x) I'd say Spring Boot 3.3 is not optional, if Grails 7 is expected to be used in any new real world production application.

**Comment by gsartori** [comment]
My concerns about building Grails on top of old technologies is not related to performance, it’s related to security and “staying relevant”. CTOs would not choose an outdated tech because of security. Developers don’t want to work on old tech, this is especially true for junior devs. So keep Grails updated to the latest libraries is a top priority in my opinion.

**Comment by rainboyan** [comment]
When we talk about supported versions of Spring Boot 3, I think there are at least three layers of implications, firstly Spring Framework 6 support because Grails actually relies on a lot of Spring Framework APIs, secondly third-party dependencies of Spring Boot, and thirdly Groovy 4 and Java 17.

If Grails 7 can support Spring Boot 3.0, it will be easy for 3.1 to 3.3.

**Comment by scrain** [comment]
Given Spring Boot 2.7.x was out of mainstream support last November, I feel that 3.3 is a must have.  https://spring.io/projects/spring-boot#support

**Comment by bkoehm** [comment]
I voted "Must have" in the poll but I wanted to add that, for me it's not so much which version of Grails will go to Spring Boot 3+, but rather the community facilitating paths for those who wish to get started (branches for pull requests, places to discuss, etc).  If the decision is to ultimately put Spring Boot 3+ in the Grails 8 roadmap instead of Grails 7, it seems reasonable to me to do that, as long as Spring Boot 3+ pull requests can start happening now and in parallel with Grails 7.

I don't think we should let the difficulties of GORM/Hibernate hold us back from getting started.  While I realize GORM is likely a "must-have" for any future Spring Boot 3+ release of Grails, I think some of us would be willing to test Grails/SpringBoot3+ development branches with our "GORMless" applications (my employer has some).  This would at least offer a starting place.

**Comment by boardbloke** [comment]
We have used Grail extensively in our projects, and the unfortunate situation is that our risk department want us off of it precisely because it doesn't track Spring releases quickly enough.

I think would continue to use Grails if that was improved.  As it stands unless Spring Boot 3 support comes in Grails 7, with a roadmap published soon, we'll be forced to transition to vanilla Boot.

**Comment by lbwexler** [comment]
Spring Boot 3.x is a must.  This is *all* about security, and the ability to support, use, and advocate for grails in an enterprise environment.

**Comment by sdelamo** [comment]
Thanks everyone for the feedback. [There is already a work in progress PR for updating Spring Boot](https://github.com/grails/grails-core/pull/13545/files) for Grails 7.
